cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -951,28 +953,27 @@ class AsofJoinNode : public ExecNode {
   }
 
   static arrow::Result<std::shared_ptr<Schema>> MakeOutputSchema(
-      const std::vector<ExecNode*>& inputs,
+      const std::vector<std::shared_ptr<Schema>> input_schema,
```

Comment:
Why this change?

Reply:
This allows exposing `MakeOutputSchema` to a caller that has a schema rather than an `ExecNode`, as is done in the tester.

Reply:
kk sounds reasonable